app.rb

```diff
@@ -36,44 +35,47 @@ def list_people
   end
 
   def create_person
-    puts 'Do you want to  create a student (1) or a teacher(2)? [Input the number 1 or 2]: '
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number 1 or 2]: '
     input_result = gets.chomp.to_i
 
     case input_result
     when 1
       create_student
     when 2
       create_teacher
+    else
+      puts 'Invalid input. Please select 1 for student or 2 for teacher.'
     end
   end
 
   def create_student
-    print 'Age:'
+    print 'Age: '
     age = gets.chomp
 
-    print 'Name:'
+    print 'Name: '
     name = gets.chomp
 
-    print 'Has Parent permission? [Y/N]'
+    print 'Has Parent permission? [Y/N]: '
     permission = gets.chomp
 
     student = Student.new(age, name, parent_permission: permission)
-    @people_list.push(student)
-    puts 'Student Created Successfully'
+    @people_list << student
+    puts 'Student created successfully'
   end
 
   def create_teacher
-    print 'Age:'
+    print 'Age: '
     age = gets.chomp
 
-    print 'Name:'
+    print 'Name: '
     name = gets.chomp
 
-    print 'Specialization:'
+    print 'Specialization: '
     specialization = gets.chomp
 
-    @people_list << Teacher.new(age, name, specialization: specialization)
-    puts 'Teacher Created Successfully'
+    teacher = Teacher.new(age, name, specialization: specialization)
+    @people_list << teacher
+    puts 'Teacher created successfully'
```

Comment:
- [ ] this is what the previous CR means with:
 _"Other logics such as requests for input (get.chomps) or input validations\print statements should be placed in the main.rb, separate classes or methods."_
In this case I suggest you create a separate class (named Input for example) with the methods person_input, book_input, rental_input and however many you think you need and put all the `print` and `gets` statements in these methods. after that in the App class you just call these methods instead of printing, getting and processing this data directly in this class.